### PR TITLE
external workload: Increase the VM creation retry count

### PR DESCRIPTION
**Pull request by michi-covalent**
Increase the retry count from 3 to 10. I have seen 3 consecutive
failures a couple of times 😐

Ref: https://github.com/cilium/cilium-cli/actions/runs/1113449756

Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

### CODE REVIEW

**Review by nbusseneau — APPROVED** [review]
:/

**Comment by michi-covalent** [comment]
rebased to pick up https://github.com/cilium/cilium-cli/pull/474

File: .github/workflows/externalworkloads.yaml
Comment (pchaigno) [inline .github/workflows/externalworkloads.yaml:77]: If 3+ retries are needed from time to time only, 10 seems a bit excessive no? Also, won't we hit the timeout of 20 minutes anyway before reaching 10 retries?

Comment (michi-covalent) [reply .github/workflows/externalworkloads.yaml:77]: timeout is per attempt https://github.com/nick-invision/retry#timeout_minutes

not sure what the right value is tbh.

Comment (pchaigno) [reply .github/workflows/externalworkloads.yaml:77]: I was referring to the timeout of the job, on line 38 of the workflow.

Comment (michi-covalent) [reply .github/workflows/externalworkloads.yaml:77]: ah yes of course. personally i'd say we might as well retry until the 20 minute timeout until we come up with a better way to handle this.

**Review by pchaigno — APPROVED** [review]
LGTM once second commit is removed of course.

**Comment by michi-covalent** [comment]
ignoring https://github.com/cilium/cilium-cli/pull/473/checks?check_run_id=3285027333, after this pr gets merged, the retry count will get bumped up.

**Pull request merged**
